dql/history.py

```diff
@@ -0,0 +1,49 @@
+import os
+from pathlib import Path
+
+
+class HistoryManager(object):
+    _initial_history_length = 0
+
+    def _create_dir_if_not_exists(self, path: str) -> None:
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+    def _create_file_if_nost_exists(self, path: str) -> None:
+        if not os.path.exists(path):
+            with open(path, "w"):
+                pass
+
+    def _get_history_dir_and_file(self):
```

Comment:
I think this is doing too much for a "get" method: it has side effects like creating directories and writing to a file. You really only need to create those in the `try_to_write` method, because `try_to_read` can just do a simple existence check on the history_file. This means we can get rid of the helper methods above (since they can be inlined) and greatly simplify this function